usaspending_api/broker/management/commands/load_fpds_transactions.py

```diff
@@ -0,0 +1,158 @@
+            lines = file.readlines(CHUNK_SIZE)
+            lines = [line.decode("utf-8") for line in lines]
+            if len(lines) == 0:
+                break
+            yield lines
+
+    def load_fpds_from_file(self, file_path):
+        with RetrieveFileFromUri(file_path).get_file_object() as file:
+            for next_batch in self.next_file_batch_generator(file):
+                id_list = [int(re.search(r"\d+", x).group()) for x in next_batch]
+                logger.info(
+                    "Loading next batch from provided file (size: {}, ids {}-{})...".format(
+                        len(id_list), id_list[0], id_list[-1]
+                    )
+                )
+                self.modified_award_ids.extend(load_ids(id_list))
+
+    def add_arguments(self, parser):
+        mutually_exclusive_group = parser.add_mutually_exclusive_group(required=True)
+
+        mutually_exclusive_group.add_argument(
+            "--ids",
+            nargs="+",
+            type=int,
+            help="Load/Reload transactions using this detached_award_procurement_id list (space-separated)",
+        )
+        mutually_exclusive_group.add_argument(
+            "--date",
+            dest="date",
+            type=datetime_command_line_argument_type(naive=True),  # Broker date/times are naive.
+            help="Load/Reload all FPDS records from the provided datetime to the script execution start time.",
+        )
+        mutually_exclusive_group.add_argument(
+            "--since-last-load",
+            action="store_true",
+            help="Equivalent to loading from date, but date is drawn from last update date recorded in DB",
+        )
+        mutually_exclusive_group.add_argument(
+            "--file",
+            metavar="FILEPATH",
+            type=str,
+            help="Load/Reload transactions using the detached_award_procurement_id list stored at this file path (one ID per line)"
+            "to reload, one ID per line. Nonexistent IDs will be ignored.",
+        )
+        mutually_exclusive_group.add_argument(
+            "--reload-all",
+            action="store_true",
+            help="Script will load or reload all FPDS records in broker database, from all time",
+        )
+
+    def handle(self, *args, **options):
+
+        # loads can take a while, so we record last updated date from the start of all transactions
+        update_time = datetime.now(timezone.utc)
+
+        if options["reload_all"]:
+            self.load_fpds_from_date(None)
+
+        elif options["date"]:
+            self.load_fpds_from_date(options["date"])
+
+        elif options["ids"]:
+            self.modified_award_ids.extend(load_ids(options["ids"]))
+
+        elif options["file"]:
+            self.load_fpds_from_file(options["file"])
+
+        elif options["since_last_load"]:
+            self.load_fpds_from_date(get_last_load_date("fpds"))
+
+        if options["reload_all"] or options["since_last_load"]:
```

Comment:
updating the last_load_date might as well occur after the awards are updated and the orphaned rows are purged.

Reply:
Keep in mind that the later steps might fail, and this script doesn't use a `finally` block. That said, now that the later steps only use the IDs put in place by the transaction loader, perhaps it would be best to NOT update since_last_load if the award updates fail, so that those touched awards get noted again.
I don't care about the orphan pruning, since that effects all rows every time, and doesn't leave the data in a bad state if it doesn't happen for a day.

Reply:
Either has tradeoffs. If a massive load of transactions fail on the award update stage, perhaps it would be beneficial to manually carry the operations to completion and directly update the date.

For smaller loads on an award update failure, re-running the script using the same params might be easier for all parties.

Leaving orphan pruning for after the date update makes sense.

Reply:
In both of the cases above, it seems like it would be better for last_load_date to not be updated. I'll move it